File: mobile/lib/domain/sponsors.dart
class Sponsors {
  Map<String, dynamic> description;
  String website;
  String imageUrl;
  SponsorDetails details;

  Sponsors({
    this.description,
    this.website,
    this.imageUrl,
    this.details
  });

  Sponsors.fromMap(Map<String, dynamic> map) {
    description = map['description'];
    website = map['website'];
    imageUrl = map['imageUrl'];
    details = SponsorDetails.fromMap(map['mobile']);
  }
}

class SponsorDetails {
    List<dynamic> padding;
    double widthFactor;
    double heightFactor;

    SponsorDetails.fromMap(Map<String, dynamic> map) {
        padding = map['padding'];
        widthFactor = map['widthFactor']?.toDouble();
        heightFactor = map['heightFactor']?.toDouble();
    }
}

File: dashboard/src/app/api/models/sponsors.ts
export interface SponsorDetails {
    padding: number[];
    widthFactor: number;
    heightFactor: number;
}

export interface Sponsors {
    _id?: string;
    name: string;
    description: { [lang: string]: string };
    website: string;
    imageUrl: string;
    web: SponsorDetails;
    mobile: SponsorDetails;
}
